### ISSUE TITLE
Investigate Observers

### ISSUE THREAD

**Issue by mrchantey** [mrchantey/beet#4]
Bevy now has [observers](https://github.com/bevyengine/bevy/blob/main/crates/bevy_ecs/src/observer/mod.rs), this may be a better approach than change detection on sparse set tag components.

- Maybe easier to reason about in selectors?
- Instant effects, means no one frame delay

## Option 1: Triggers for `RunResult` (partial adoption) 

We could use trigger add/remove systems that run on some query like:

```rs
fn add_child_triggers(commands: Commands, parents: Query<&Children, Added<Running>>){
  for (parent, children) in parents.iter(){
    for child in children{
      commands.entity(parent).add_trigger::<RunResult>(child)
    }
  }
}
```
And i suppose something similar when it stops running..

## Option 2: Triggers for `RunResult` and `Running`

Whenever children are added to a parent:
1. the child observes the parent 'StopRunning' and also stops running as a result
2. the parent observes the child 'StopRunning' and makes some decision or bubbles it up..
3. the parent can trigger 'StartRunning' on the child

```

```

**Comment by mrchantey** [comment]
done, fully integrated

**Comment by mgi388** [comment]
@mrchantey nice, is there a commit that fixes this? Can't see anything in https://github.com/mrchantey/beet/commits/main/

**Comment by mrchantey** [comment]
yeah sry should've closed with a pr, here it is #6
